OptimizelySDK/Optimizely.cs

```diff
@@ -188,19 +237,64 @@ public static String SDK_TYPE
             EventBuilder = new EventBuilder(Bucketer, Logger);
             UserProfileService = userProfileService;
             NotificationCenter = notificationCenter ?? new NotificationCenter(Logger);
-            DecisionService = new DecisionService(Bucketer, ErrorHandler, userProfileService, Logger);
-            EventProcessor = eventProcessor ?? new ForwardingEventProcessor(EventDispatcher, NotificationCenter, Logger);
-            DefaultDecideOptions = defaultDecideOptions ?? new OptimizelyDecideOption[] { };
+            DecisionService =
+                new DecisionService(Bucketer, ErrorHandler, userProfileService, Logger);
+            EventProcessor = eventProcessor ?? new ForwardingEventProcessor(EventDispatcher,
+                NotificationCenter,
+                Logger);
+            DefaultDecideOptions = defaultDecideOptions ?? new OptimizelyDecideOption[]
+                { };
+        }
+
+#if USE_ODP
+        private void SetupOdp(ProjectConfig config, OptimizelySdkSettings sdkSettings = null)
+        {
+            if (config == null)
+            {
+                return;
+            }
+
+            SdkSettings = sdkSettings ?? new OptimizelySdkSettings();
+
+            var odpConfig = new OdpConfig(config.PublicKeyForOdp, config.HostForOdp,
+                config.Segments.ToList());
+
+            OdpManager = new OdpManager.Builder().
+                WithOdpConfig(odpConfig).
+                WithCacheSize(SdkSettings.SegmentsCacheSize).
+                WithCacheTimeout(SdkSettings.SegmentsCacheTimeout).
+                WithLogger(Logger).
+                WithErrorHandler(ErrorHandler).
+                Build(!SdkSettings.DisableOdp);
+
+            NotificationCenter.AddNotification(
```

Comment:
Use `inline` delegate instead of separate class method.

Reply:
This actually saved part of the problem I was trying to handle with `GetCurrentProjectConfig()`...but let's discuss.